src/Client.php

```diff
@@ -384,9 +448,17 @@ public function sendMessage(Message $message)
    * @param \Maknz\Slack\Message $message
    * @return void
    */
-  public function queueMessage(Message $message)
+  public function queueMessage(Message $message, $numRetries = self::MAX_RETRY_ATTEMPTS)
```

Comment:
I don't think we need a default value for `$numRetries` here. We are/should be handling it before the flow reaches this function. 
I think all the functions in this class are for internal use only and that the client would be using the `Message` class, where we should be handling these kind of validations too.
What do you think?